docs/source/raft_configuration.md

```diff
@@ -330,5 +330,34 @@ channels (this does not mean they share data --- they do not!), if a Raft node i
 part of the consenter set in many channels, you might want to lengthen the amount
 of time it takes to trigger an election to avoid inadvertent leader elections.
 
+2. As aforementioned, write-ahead logs (wal) and snapshots should be persisted in
+order to attain safety property of Raft protocol. However, it may happen if a node
+loses its data by accident, and the node would panic with following message:
+```
+tocommit(X) is out of range [lastIndex(Y)]. Was the raft log corrupted, truncated, or lost?
+```
+Manual operation would be required to resurrect problematic node in this case.
+* Safe recovery:  
+The only way to handle the corruption or loss of a WAL is to remove the corrupted
+consenter from the quorum, and add a new one in its place, which can be done following
+instructions in reconfiguration section in this doc. This is the most appropriate
+option, as it does not violate safety property of Raft protocol. However, this is heavy
+operation and affects all channels this node participates in.
```

Comment:
Again, not sure what "safety property" means here. Do you mean something like "does not violate the safe operation of a Raft ordering service" or "does not violate the integrity of the ordering service"?